lib/Matrix.js

```diff
@@ -138,7 +142,20 @@ function useMatrixProvider(auth) {
             }
         }
         if ((membership === 'invite')) {
-            setInvites(setRoom(room.roomId));
+            // we need to update the matrixClient room sync so getRooms(roomId) in buildRoomObject doesn't return null
+            await matrixClient.roomInitialSync(room.roomId)
+                .catch(() => {});
+            // it seems that invitations don't list custom state events, therefor we need to fetch dev.medienhaus.meta manually
+            const metaEvent = await hydrateMetaEvent(room.roomId);
+            // we check to see who invited the current user
+            const inviterName = room.getMember(matrixClient.getUserId())?.events?.member?.getSender?.();
+            const inviter = matrixClient.getUser(inviterName);
+            setInvites(setRoom(room.roomId, metaEvent, inviter));
+        }
+
+        if (prevMembership === 'invite') {
+            // in this case we know the invitation was handled by the user and we can delete it from our invitations
+            setInvites(deleteElement(room.roomId));
```

Comment:
Isn't line `137` taking care of this already? If not, why not and in which case?

Reply:
ah yeah, I think this was left over from me trying to fix the weird issue with invitations reappearing.